ParameterDataApi.yaml

```diff
@@ -88,122 +144,79 @@ components:
+      
+        messageType:
+          type: string          
+          description: The type of the message, dictated by the client for delineating individual message formats
+
+        messageVersion:
           type: string
-          format: date-time
+          description: The version string for the parameter data messageType. If the client changes the data fields within a message stream, they should increment the version accordingly to distinguish message format along with the messageType
           
-        columns:
-          $ref: '#/components/schemas/ColumnDefinition'
-        
       required:
-        - startTime
-        - endTime
-        - columns
+        - componentUuid
+        - messageType
+        - version
 
-    TimeSeriesDataDefinition:
+    SingleDataDefinition:
       type: object
-      description: A definition of time series data. This is loosely based upon Json Time Series, Version 1.0. (https://docs.eagle.io/en/latest/reference/index.html)
+      description: A definition of a set of parameter data having a single discrete timestamp. For multiple parameter observations for the given timestamp, use multiple key/value pairs with unique keys
       properties:
-        docType:
-          type: string          
-          description: The type of the document
-
-        version:
-          type: string
-          description: The version string for the time series data definition
 
         header:
           $ref: '#/components/schemas/HeaderDefinition'
 
-        data:
-          $ref: '#/components/schemas/DataDefinition'
-
+        single:
+          $ref: '#/components/schemas/DataSample'
+         
       required: 
-        - docType
         - header
-        - data
+        - single
 
-    ValueSpec:
+    TimeSeriesDataDefinition:
       type: object
+      description: A time series ordered collection for a set of parameter data. For multiple parameter observations for the given timestamp, use multiple key/value pairs with unique keys. It is encouraged that each element in the array has the same keys in the same order
       properties:
-        value:
-          description: The value of the data sample described in string form. 
-          type: string
-          
-        quality: 
-          description: The quality score of the data sample (should be bounded 0.0 .. 100.0)
-          type: number          
-          format: double
-          minimum: 0.0
-          maximum: 100.0
-        
-        annotation: 
-          description: The annotation for the data sample
-          type: string
-          
-      required:
-        - value
+
+        header:
+          $ref: '#/components/schemas/HeaderDefinition'
+
+        series:
+          type: array
+          items: 
+            $ref: '#/components/schemas/DataSample'
+                        
+      required: 
+        - header
+        - series
         
   securitySchemes:
```

Comment:
General Comment: Do we eventually need the same admin role here as was implemented elsewhere recently?

Reply:
If you are referring to the stuff Steven was working on, no. These pushes are 'fire and forget' rather than CRUD operations on individual resources. The admin role there was to help maintain ownership over individual resources. Here, clients are just dumping data.